MetaMask/metamask-extension
# Part of #17670 for file signature-request-data.js

## Explanation
This PR is a part of the issue #17670 Replace deprecated Typography with Text component for the file 
**ui\components\app\signature-request\signature-request-data\signature-request-data.js**



## Screenshots/Screencaps



### Before

<img width="828" alt="Screenshot 2023-07-24 at 12 30 00 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/fa334b20-915b-496a-9c4a-2e98469eca46">


### After

<img width="873" alt="Screenshot 2023-07-24 at 12 29 46 PM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/fad214e8-1d32-4393-9b61-af49a7f8115d">


https://github.com/MetaMask/metamask-extension/assets/8112138/7f022364-9937-4708-9bb5-5174fcd59ef9




## Manual Testing Steps
- Pull this branch
- Run storybook `yarn storybook`
- Check `SignatureRequestData` story against https://metamask.github.io/metamask-storybook/?path=/story/components-app-signaturerequest-signaturerequestdata--default-story
OR
- use the test dapp page https://metamask.github.io/test-dapp/
- Scroll down to Sign Typed Data V4
- Click SIGN
- See `SignatureRequestData`


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
